app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/LearningActivity.kt

```diff
@@ -1,24 +1,125 @@
 import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.model.DummyDatasetInterface
+import com.github.HumanLearning2021.HumanLearningApp.presenter.DummyUIPresenter
 import com.github.HumanLearning2021.HumanLearningApp.presenter.LearningPresenter
+import kotlinx.coroutines.launch
 
 class LearningActivity : AppCompatActivity() {
+
+    private val dummyPres = DummyUIPresenter(DummyDatasetInterface())
+    private val learningPresenter = LearningPresenter(DummyDatasetInterface())
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_learning)
 
-        findViewById<ImageView>(R.id.learning_cat_0)
-                .setOnDragListener(LearningPresenter.targetOnDragListener)
-        findViewById<ImageView>(R.id.learning_cat_1)
-                .setOnDragListener(LearningPresenter.targetOnDragListener)
-        findViewById<ImageView>(R.id.learning_cat_2)
-                .setOnDragListener(LearningPresenter.targetOnDragListener)
+        lifecycleScope.launch {
+            val cats = DummyDatasetInterface().getCategories()
+            if (BuildConfig.DEBUG && cats.size < 3) {
+                // TODO : maybe allow fewer categories in the future
+                error("There should be at least 3 categories")
+            }
+
+            val cat0Name = cats.elementAt(0).name
+
+            initTargetCategory(R.id.learning_cat_0, cat0Name)
+            initTargetCategory(R.id.learning_cat_1, cats.elementAt(1).name)
+            initTargetCategory(R.id.learning_cat_2, cats.elementAt(2).name)
+
+            initImageToSort(R.id.learning_im_to_sort, cat0Name)
+        }
+
+
+    }
+
+    private fun initImageView(catIvId: Int, catName: String): ImageView {
+        val catIv = findViewById<ImageView>(catIvId)
+        catIv.contentDescription = catName
+        lifecycleScope.launch {
+            dummyPres.getPicture(catName)?.displayOn(this@LearningActivity, catIv)
+        }
+        return catIv
+    }
+
+    private fun initImageToSort(catIvId: Int, catName: String) {
+        initImageView(catIvId, catName).setOnTouchListener(LearningActivity::onImageToSortTouched)
+    }
+
+    private fun initTargetCategory(catIvId: Int, catName: String) {
+        initImageView(catIvId, catName).setOnDragListener(targetOnDragListener)
+    }
+
+    private val opaque = 1.0f
+    private val halfOpaque = opaque / 2
+    private val targetOnDragListener = View.OnDragListener { v, event ->
+        when (event.action) {
+            DragEvent.ACTION_DRAG_ENTERED -> dragEnteredCallback(v)
+            DragEvent.ACTION_DRAG_EXITED -> dragExitedCallback(v)
+            DragEvent.ACTION_DROP -> dropCallback(event, v)
+
+            // DragEvent.ACTION_DRAG_ENDED &
+            // DragEvent.ACTION_DRAG_LOCATION & DragEvent.ACTION_DRAG_STARTED
+            else -> true
```

Comment:
What is the purpose of the commented lines here?

Reply:
It's just to remember which cases go in the else branch, if I ever need to handle those cases in the future.